includes/class-sensei-grading.php

```diff
@@ -1124,13 +1124,47 @@ public static function get_graded_lessons_sum() {
 		$comment_query_piece['from']   = " FROM {$wpdb->comments}  INNER JOIN {$wpdb->commentmeta}  ON ( {$wpdb->comments}.comment_ID = {$wpdb->commentmeta}.comment_id ) ";
 		$comment_query_piece['where']  = " WHERE {$wpdb->comments}.comment_type IN ('sensei_lesson_status') AND ( {$wpdb->commentmeta}.meta_key = 'grade')";
 
-		$comment_query     = $comment_query_piece['select'] . $comment_query_piece['from'] . $comment_query_piece['where'];
+		$comment_query = $comment_query_piece['select'] . $comment_query_piece['from'] . $comment_query_piece['where'];
+		// WPCS: unprepared SQL OK.
 		$sum_of_all_grades = intval( $wpdb->get_var( $comment_query, 0, 0 ) );
 
 		return $sum_of_all_grades;
 
 	}
 
+	/**
+	 * Get average grade of all lessons graded in all the courses.
+	 *
+	 * @since 4.2.0
+	 * @access public
+	 * @return double $graded_lesson_average_grade Average value of all the graded lessons in all the courses.
+	 */
+	public static function get_graded_lessons_average_grade() {
+		// Fetching all the grades of all the lessons that are graded.
+		global $wpdb;
+
+		$comment_query_piece           = [];
+		$comment_query_piece['select'] = "SELECT {$wpdb->commentmeta}.meta_value AS grades";
+		$comment_query_piece['from']   = " FROM {$wpdb->comments}  INNER JOIN {$wpdb->commentmeta}  ON ( {$wpdb->comments}.comment_ID = {$wpdb->commentmeta}.comment_id ) ";
+		$comment_query_piece['where']  = " WHERE {$wpdb->comments}.comment_type IN ('sensei_lesson_status') AND ( {$wpdb->commentmeta}.meta_key = 'grade')";
+
+		$comment_query = $comment_query_piece['select'] . $comment_query_piece['from'] . $comment_query_piece['where'];
+		// WPCS: unprepared SQL OK.
+		$all_grades    = $wpdb->get_results( $comment_query );
+		$all_grades    = is_array( $all_grades ) ? $all_grades : [ $all_grades ];
```

Comment:
Good solution! Just a point, will getting the sum and the count directly from the DB be okay for us here? I was thinking about using the sum and count function like this 
```
$comment_query_piece           = [];
$comment_query_piece['select'] = "SELECT SUM( {$wpdb->commentmeta}.meta_value ) AS grade_sum, COUNT( * ) as grade_count";
$comment_query_piece['from']   = " FROM {$wpdb->comments}  INNER JOIN {$wpdb->commentmeta}  ON ( {$wpdb->comments}.comment_ID = {$wpdb->commentmeta}.comment_id ) ";
$comment_query_piece['where']  = " WHERE {$wpdb->comments}.comment_type IN ('sensei_lesson_status') AND ( {$wpdb->commentmeta}.meta_key = 'grade')";

$comment_query = $comment_query_piece['select'] . $comment_query_piece['from'] . $comment_query_piece['where'];
// WPCS: unprepared SQL OK.
$sum_result    = $wpdb->get_row( $comment_query );
return $sum_result->grade_sum / $sum_result->grade_count;
```

Reply:
@Imran92  Thank is a very good solution will use it. Thankxxx

Reply:
Here is the [fix ](https://github.com/Automattic/sensei/pull/4865/commits/031b6edb4ef43ab66b6885b8f36c549b0f320862)